django/contrib/admin/widgets.py

```diff
@@ -384,10 +384,12 @@ def __init__(self, rel, admin_site, attrs=None, choices=(), using=None):
         self.db = using
         self.choices = choices
         self.attrs = {} if attrs is None else attrs.copy()
+        self.to_field = self.rel.field.to_fields[0] if hasattr(self.rel.field, 'to_fields') else 'pk'
 
     def get_url(self):
         model = self.rel.model
-        return reverse(self.url_name % (self.admin_site.name, model._meta.app_label, model._meta.model_name))
+        url = reverse(self.url_name % (self.admin_site.name, model._meta.app_label, model._meta.model_name))
+        return url + '?to_field=' + self.to_field
```

Comment:
You can avoid adding `to_field` if not necessary by making the `self.to_field` assignment default to `None` if not present and only appending it if necessary here.